delivery_postlogistics/postlogistics/web_service.py

```diff
@@ -0,0 +1,497 @@
+        postlogistics_partner_codes = partner.postlogistics_option_ids.mapped("code")
+
+        is_phone_required = "ZAW3213" in postlogistics_partner_codes
+
+        if is_phone_required:
+            phone = picking.delivery_phone or partner.phone
+            if phone:
+                recipient["phone"] = phone
+
+            mobile = picking.delivery_mobile or partner.mobile
+            if mobile:
+                recipient["mobile"] = mobile
+
+        return recipient
+
+    def _prepare_customer(self, picking):
+        """ Create a ns0:Customer as a dict from picking
+
+        This is the Postlogistic Customer, thus the sender
+
+        :param picking: picking browse record
+        :return a dict containing data for ns0:Customer
+
+        """
+        company = picking.company_id
+        partner = company.partner_id
+
+        customer = {
+            "name1": partner.name,
+            "street": partner.street,
+            "zip": partner.zip,
+            "city": partner.city,
+            "country": partner.country_id.code,
+            "domicilePostOffice": picking.carrier_id.postlogistics_office,
+        }
+        logo = picking.carrier_id.postlogistics_logo
+        if logo:
+            logo_image = Image.open(BytesIO(base64.b64decode(logo)))
+            logo_format = logo_image.format
+            customer["logo"] = logo.decode()
+            customer["logoFormat"] = logo_format
+        return customer
+
+    def _get_label_layout(self, picking):
+        """
+        Get Label layout define in carrier
+        """
+        return picking.carrier_id.postlogistics_label_layout.code
+
+    def _get_output_format(self, picking):
+        """
+        Get Output format define in carrier
+        """
+        return picking.carrier_id.postlogistics_output_format.code
+
+    def _get_image_resolution(self, picking):
+        """
+        Get Output Resolution Code define in carrier
+        """
+        return picking.carrier_id.postlogistics_resolution.code
+
+    def _get_license(self, picking):
+        """ Get the license
+
+        Take it from carrier and if not defined get the first license.
+
+        :return: license number
+        """
+        franking_license = picking.carrier_id.postlogistics_license_id
+        return franking_license.number
+
+    def _prepare_attributes(self, picking, pack=None, pack_num=None, pack_total=None):
+        packaging = (
+            pack
+            and pack.packaging_id
+            or picking.carrier_id.postlogistics_default_packaging_id
+        )
+        services = packaging._get_packaging_codes()
+
+        total_weight = pack.shipping_weight if pack else picking.shipping_weight
+        total_weight *= 1000
+
+        if not services:
+            raise exceptions.UserError(
+                _("No Postlogistics packaging services found in this picking.")
+            )
+
+        attributes = {
+            "weight": total_weight,
```

Comment:
@jbaudoux, about the max weight, last time I check, the error will be raised when the weight exceed 30kg (which is 30,000g). Btw, I dont think we need to put a validation here, because when whe push the package, and if there is an error raised from the server, our odoo will raise an exception for that.
For the type of the weight, I will force-format it in integer type

Reply:
@phuctranerp ok. Thanks